File: 2401.cpp
// Ven primo
// https://jv.umsa.bo/problem.php?id=2401

#include <iostream>
#include <vector>
#include <cmath>
#include <algorithm>

using namespace std;

bool es_primo(int num) {
    if (num < 2)
        return false;
    if (num == 2 || num == 3)
        return true;
    if (num % 2 == 0 || num % 3 == 0)
        return false;
    int limite = static_cast<int>(sqrt(num)) + 1;
    for (int i = 5; i < limite; i += 6) {
        if (num % i == 0 || num % (i + 2) == 0)
            return false;
    }
    return true;
}

int main() {
    int n;
    cin >> n;

    vector<int> ls;
    while (ls.size() < n) {
        int num;
        cin >> num;
        if (es_primo(num)) {
            ls.push_back(num);
        }
    }
    sort(ls.begin(), ls.end());
    for (int elem : ls) {
        cout << elem << " ";
    }
    cout << endl;
    return 0;
}

File: jvPY/2401.py
# Ven Primo
# https://jv.umsa.bo/problem.php?id=2401


import math

def es_primo(num):
    if num < 2:
        return False
    if num == 2 or num == 3:
        return True
    if num % 2 == 0 or num % 3 == 0:
        return False

    limite = int(math.sqrt(num)) + 1
    for i in range(5, limite, 6):
        if num % i == 0 or num % (i + 2) == 0:
            return False

    return True

n = int(input())
ls = []
while len(ls) < n:
    num = int(input())
    if es_primo(num):
        ls.append(num)
res = sorted(ls)
for elem in res:
    print(str(elem)+" ",end="")

#  4, 12, 17, 2, 25, 36, 3, 9, 12, 23, 40, 11, 17